src/http2_frame_headers.erl

```diff
@@ -4,40 +4,73 @@
 
 -behaviour(http2_frame).
 
--export([
+-export(
+   [
     format/1,
     from_frames/1,
+    new/1,
+    new/2,
     read_binary/2,
     to_frames/5,
     to_binary/1
   ]).
 
--spec format(headers()) -> iodata().
+-record(headers,
+        {
+          priority = undefined :: http2_frame_priority:payload() | undefined,
+          block_fragment :: binary()
+        }).
+-type payload() :: #headers{}.
+-export_type([payload/0]).
+
+-spec format(payload()) -> iodata().
 format(Payload) ->
     io_lib:format("[Headers: ~p]", [Payload]).
 
--spec read_binary(binary(), frame_header()) ->
+-spec new(binary()) -> payload().
```

Comment:
I sort of feel like this module should handle the hpack abstraction for us, but maybe that's not possible with all of the context needed.